### PR TITLE
Aswathy/Personal details

### PR DESCRIPTION
**Pull request by aswathy-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/pages/signup-affiliates/components/_personal-details.tsx [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+        border-bottom: 1px solid;
+        margin: auto;
+        color: var(--color-grey-8);
+    }
+    span::before {
+        margin-right: 8px;
+    }
+    span::after {
+        margin-left: 8px;
+    }
```
[the remark this answers is not in the document]
Comment (aswathy-deriv) [reply src/pages/signup-affiliates/components/_personal-details.tsx:67]: okay

File: src/pages/signup-affiliates/components/_personal-details.tsx [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+    const [social_media_url_error_msg, setSocialMediaErrorMsg] = useState()
+    const [password_error_msg, setPasswordErrorMsg] = useState()
+
+    const form_inputs = [
+        {
+            id: 'first_name',
+            name: 'first_name',
+            type: 'text',
+            label: localize('First name'),
+            placeholder: 'First name',
```
[the remark this answers is not in the document]
Comment (aswathy-deriv) [reply src/pages/signup-affiliates/components/_personal-details.tsx:132]: okay

File: src/pages/signup-affiliates/components/_personal-details.tsx [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+            required: true,
+            value_set: setFirstName,
+            error_set: setFirstNameErrorMsg,
+        },
+        {
+            id: 'last_name',
+            name: 'last_name',
+            type: 'text',
+            label: localize('Last name'),
+            placeholder: 'First name',
```
[the remark this answers is not in the document]
Comment (aswathy-deriv) [reply src/pages/signup-affiliates/components/_personal-details.tsx:145]: okay

File: src/pages/signup-affiliates/components/_personal-details.tsx [hunk 4]
```diff
@@ -0,0 +1,519 @@
...
+
+    const { send } = useDerivWS()
+
+    const getCitizenList = () => {
+        return {
+            residence_list: 1,
+        }
+    }
+
+    const citizen_list = getCitizenList()
```
[the remark this answers is not in the document]
Comment (aswathy-deriv) [reply src/pages/signup-affiliates/components/_personal-details.tsx:264]: okay

[the remark this answers is not in the document]
Comment (aswathy-deriv) [reply src/pages/signup-affiliates/components/_personal-details.tsx:264]: okay

File: src/components/form/uploader.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+const UploadLabel = styled(StyledLabel)`
+    width: 70%;
+    line-height: 15px;
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+`
+
+const ExtraInfo = styled.div<{ p?: string }>`
```
Comment (prince-deriv) [inline src/components/form/uploader.tsx:40]: are there any props passed here?

File: src/pages/signup-affiliates/components/_personal-details.tsx [hunk 6]
```diff
@@ -0,0 +1,545 @@
...
+            certificate_of_incorporation,
+            citizen,
+            website_url,
+            social_media_url,
+            password,
+            currency,
+        })
+    }, [personal_details])
+
+    const is_individual_validate =
```
Comment (prince-deriv) [inline src/pages/signup-affiliates/components/_personal-details.tsx:317]:
```suggestion
    const is_valid_individual =
```

File: src/pages/signup-affiliates/components/_personal-details.tsx [hunk 7]
```diff
@@ -0,0 +1,545 @@
...
+        !first_name || !last_name || !date_birth || !citizen || !password || !currency
+
+    const is_business_validate =
```
Comment (prince-deriv) [inline src/pages/signup-affiliates/components/_personal-details.tsx:320]:
```suggestion
    const is_valid_business =
```

File: src/pages/signup-affiliates/components/_personal-details.tsx [hunk 8]
```diff
@@ -0,0 +1,545 @@
...
+        !citizen ||
+        !password ||
+        !company_name ||
+        !company_registration_number ||
+        !currency ||
+        !certificate_of_incorporation
+
+    useEffect(() => {
+        const validateAccountType = is_individual ? !is_individual_validate : !is_business_validate
+        onValidate(validateAccountType)
```
Comment (prince-deriv) [inline src/pages/signup-affiliates/components/_personal-details.tsx:333]:
```suggestion
        const is_valid_account = is_individual ? !is_individual_validate : !is_business_validate
        onValidate(is_valid_account)
```

File: src/pages/signup-affiliates/components/_personal-details.tsx [hunk 9]
```diff
@@ -0,0 +1,545 @@
...
+            case 'password': {
+                setPersonalDetails({ ...personal_details, password: value })
+                return setErrorMsgs({ ...error_msgs, password: validation.password(value) })
+            }
+            case 'currency': {
+                setPersonalDetails({ ...personal_details, currency: value })
+                break
+            }
+        }
+    }
```
Comment (prince-deriv) [inline src/pages/signup-affiliates/components/_personal-details.tsx:409]: Can possibly be handled in one condition by using item.name

File: src/pages/signup-affiliates/components/_personal-details.tsx (hunk 6 above)
Comment (aswathy-deriv) [reply src/pages/signup-affiliates/components/_personal-details.tsx:317]: okay

File: src/pages/signup-affiliates/components/_personal-details.tsx (hunk 7 above)
Comment (aswathy-deriv) [reply src/pages/signup-affiliates/components/_personal-details.tsx:320]: okay

File: src/pages/signup-affiliates/components/_personal-details.tsx (hunk 8 above)
Comment (aswathy-deriv) [reply src/pages/signup-affiliates/components/_personal-details.tsx:333]: okay

**Pull request merged**